### PR TITLE
[MM-34203] Send team id on bindings call

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The team id was previously not being sent on binding calls. We discussed in our team meeting that we want to send this from the client.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-34203

```release-note
NONE
```

### CODE REVIEW

File: packages/mattermost-redux/src/client/client4.ts [hunk 1]
```diff
@@ -3425,9 +3425,16 @@ export default class Client4 {
...
+    getAppsBindings = async (userID: string, channelID: string, teamID: string) => {
+        const params = {
+            user_id: userID,
+            channel_id: channelID,
+            team_id: teamID,
+            user_agent: 'webapp',
+        };
+
+        return this.doFetch(
+            `${this.getAppsProxyRoute()}/api/v1/bindings${buildQueryString(params)}`,
```
Comment (larkox) [inline packages/mattermost-redux/src/client/client4.ts:3437]: :heart_eyes:

File: packages/mattermost-redux/src/actions/apps.ts [hunk 2]
```diff
@@ -12,8 +14,13 @@ import {bindClientFunc} from './helpers';
 // susceptible to breaking changes without pushing the major version of this package.
 
 export function fetchAppBindings(userID: string, channelID: string): ActionFunc {
-    return bindClientFunc({
-        clientFunc: () => Client4.getAppsBindings(userID, channelID),
-        onSuccess: AppsTypes.RECEIVED_APP_BINDINGS,
-    });
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const channel = getChannel(getState(), channelID);
+        const teamID = channel?.team_id || '';
```
Comment (larkox) [inline packages/mattermost-redux/src/actions/apps.ts:19]: Does it make sense / is it desirable to send the current team id if the channel does not have a team id?

**Review by larkox — APPROVED** [review]

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Regression tested 2 different 
- Tested across various teams 
- No regressions found found 

LGTM!

**Comment by DHaussermann** [comment]
@amyblais would we be able to cherry-pick this (once dev review is done) and the corresponding mobile PR https://github.com/mattermost/mattermost-mobile/pull/5316 ?

**Comment by amyblais** [comment]
@mickmister @jfrerich Please help prioritize this PR.

**Pull request merged**